cmd/thv-operator/controllers/mcpregistry_controller.go

```diff
@@ -301,3 +313,53 @@ func (*MCPRegistryReconciler) applyStatusUpdates(
 
 	return nil
 }
+
+// validateAndUpdatePodTemplateStatus validates the PodTemplateSpec and updates the MCPRegistry status
+// with appropriate conditions. Returns true if validation passes, false otherwise.
+func (r *MCPRegistryReconciler) validateAndUpdatePodTemplateStatus(
```

Comment:
I'd like to refactor this function in future. I'm not a fan of modifying the statuses like the below, we should probably abstract stuff like this under a separate package to keep things clean. There is already a `mcpregistrystatus` package, perhaps we reuse that or make a generic function in future for reuse across custom resources.